ast/ast.go

```diff
@@ -620,26 +620,71 @@ type SubQuery struct {
 	Limit   *Limit   // optional
 }
 
+// StarModifierExcept is EXCEPT node in Star and DotStar of SelectItem.
+//
+//	EXCEPT ({{Columns | sqlJoin ", "}})
+type StarModifierExcept struct {
+	// pos = Except
+	// end = Rparen + 1
+
+	Except token.Pos
+	Rparen token.Pos
+
+	Columns []*Ident
+}
+
+// StarModifierReplaceItem is a single item of StarModifierReplace.
+//
+//	{{.Expr | sql}} AS {{.Name | sql}}
+type StarModifierReplaceItem struct {
+	// pos = Expr.pos
+	// end = Name.end
+
+	Expr Expr
+	Name *Ident
+}
+
+// StarModifierReplace is REPLACE node in Star and DotStar of SelectItem.
+//
+//	REPLACE ({{Columns | sqlJoin ", "}})
+type StarModifierReplace struct {
+	// pos = Replace
+	// end = Rparen + 1
+
+	Replace token.Pos
+	Rparen  token.Pos
+
+	Columns []*StarModifierReplaceItem
```

Comment:
IMO, we can reuse `Alias` here instead of introducing a new struct.

Reply:
IMO, it is better to separate between `expr AS alias` and `expr [AS] alias`.
Currently they are not compatible with `parseCommaSeparatedList` unless we have a no-arg named `parse*` function.

Reply:
It seems `StarModifierReplace` is the first structure having `expr AS name[, ...]`.
IMO, this is reason enough to have a separate `StarModifierReplaceItem` at this point.